yarn-project/circuit-types/src/interfaces/prover-client.ts

```diff
@@ -29,8 +30,11 @@ export type ProverConfig = {
 /**
  * The interface to the prover client.
  * Provides the ability to generate proofs and build rollups.
+ * TODO(palla/prover-node): Rename this interface
  */
-export interface ProverClient extends BlockProver {
+export interface ProverClient {
+  createBlockProver(db: MerkleTreeOperations): BlockProver;
```

Comment:
Many of the changes on this PR are just moving interfaces and types around just so we can reference `MerkleTreeOperations` from here in circuit-types/interfaces.